src/System.Reflection.MetadataLoadContext/ref/System.Reflection.MetadataLoadContext.cs

```diff
@@ -0,0 +1,35 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+// ------------------------------------------------------------------------------
+// Changes to this file must follow the http://aka.ms/api-review process.
+// ------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+
+namespace System.Reflection
+{
+    public abstract class MetadataAssemblyResolver
+    {
+        public abstract Assembly Resolve(MetadataLoadContext context, AssemblyName assemblyName = null);
+    }
+
+    public class PathAssemblyResolver : MetadataAssemblyResolver
+    {
+        public PathAssemblyResolver(IEnumerable<string> assemblyPaths) { }
+        public override Assembly Resolve(MetadataLoadContext context, AssemblyName assemblyName) { throw null; }
+    }
+
+    public sealed partial class MetadataLoadContext : System.IDisposable
+    {
+        public MetadataLoadContext(MetadataAssemblyResolver resolver, string coreAssemblyName = null) { }
+        public string CoreAssemblyName { get { throw null; } }
```

Comment:
As discussed offline, we should replace this with property that returns resolved CoreAssembly as Assembly

Reply:
Done